packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -148,21 +147,22 @@ export async function executePlugins(
 
   logMultipleResults(results, 'Plugins', undefined, errorsCallback);
 
-  // TODO: add groupBy method for promiseSettledResult by status, #287
-
-  const failedResults = results.filter(isPromiseRejectedResult);
-  if (failedResults.length) {
-    const errorMessages = failedResults
+  const { fulfilled, rejected } = groupByStatus(results) as {
+    fulfilled: PromiseFulfilledResult<PluginReport>[];
+    rejected: PromiseRejectedResult[];
+  };
+  if (rejected.length) {
+    const errorMessages = rejected
       .map(({ reason }: PromiseRejectedResult) => reason)
       .join(', ');
     throw new Error(
-      `Plugins failed: ${failedResults.length} errors: ${errorMessages}`,
+      `Plugins failed: ${rejected.length} errors: ${errorMessages}`,
     );
   }
 
-  return results
-    .filter(isPromiseFulfilledResult)
-    .map((result: PromiseFulfilledResult<PluginReport>) => result.value);
+  return fulfilled.map(
+    (result: PromiseFulfilledResult<PluginReport>) => result.value,
+  );
```

Comment:
Unnecessary parameter type annotation:

```suggestion
  return fulfilled.map(result => result.value);
```